### PR TITLE
Bst

### PR DESCRIPTION
**Pull request by SammyRobensParadise**

### CODE REVIEW

**Comment by SammyRobensParadise** [comment]
updated to have both recursive `print()` and `exists()`

**Comment by maryrsmcphee** [comment]
Looks good! As long as it's not going against the instructions: "For Part2, we are going to use Binary Search ADT to model and implement this prototype. Submit two files:  lab3_binary_search_tree.hpp and lab3_binary_search_tree.cpp. Do not modify the signatures for any of the functions already implemented or listed below." then I think it's fine

**Comment by maryrsmcphee** [comment]
I'm wondering if we should maybe have the recursive functions as new helper functions and then call them inside the current functions? that way we aren't changing anything with the signature? Idk

**Review by maryrsmcphee — APPROVED** [review]

**Comment by SammyRobensParadise** [comment]
> I'm wondering if we should maybe have the recursive functions as new helper functions and then call them inside the current functions? that way we aren't changing anything with the signature? Idk

thats what I did??

File: Lab3/binary_search_tree.cpp [hunk 1]
```diff
@@ -54,23 +54,22 @@ unsigned int BinarySearchTree::height() const {
...
+    if (node == NULL) {
+        return;
+    } else {
+        print(node->left);
+
+        print(node->right);
+    }
+}
+
 void BinarySearchTree::print() const {
```
Comment (SammyRobensParadise) [inline Lab3/binary_search_tree.cpp:67]: this is the original signature

File: Lab3/binary_search_tree.cpp [hunk 2]
```diff
@@ -54,23 +54,22 @@ unsigned int BinarySearchTree::height() const {
 }
 
 // PURPOSE: Prints the contents of the tree format not specified
-// TODO @Sammy
+void BinarySearchTree::print(struct BinarySearchTree::TaskItem *node) const {
```
Comment (SammyRobensParadise) [inline Lab3/binary_search_tree.cpp:57]: this is the overloaded helper

**Comment by maryrsmcphee** [comment]
LOL MY BAD u right looks great

**Comment by SammyRobensParadise** [comment]
> LOL MY BAD u right looks great

great im gonna merge

**Pull request merged**